gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ReferralRequestMapperTest.java

```diff
@@ -59,38 +62,55 @@ public class ReferralRequestMapperTest {
     @InjectMocks
     private static ReferralRequestMapper referralRequestMapper;
 
+    @Test
+    public void mapReferralRequestWithReferralRequestAsOrganization() {
+
+        var codeableConcept = createCodeableConcept(REASON_CODE_1, SNOMED_SYSTEM, CODING_DISPLAY);
+        when(codeableConceptMapper.mapToCodeableConcept(any()))
+            .thenReturn(codeableConcept);
+
+        var ehrExtract = unmarshallEhrExtractElement("full_valid_data_with_referral_request.xml");
+
+        var referralRequest = mapReferralRequest(ehrExtract,
+                                                 ehrExtract.getComponent().get(0).getEhrFolder().getComponent().get(0).getEhrComposition(),
+                                                 composition -> composition.getComponent().get(0).getRequestStatement());
+
+        assertAll(
+            () -> assertFixedValues(referralRequest),
+            () -> assertEquals(EXAMPLE_ID, referralRequest.getId()),
+            () -> assertThat(referralRequest.getAuthoredOn()).isEqualTo("2020-11-17T13:30:32:00"),
+            () -> assertEquals(PRACTITIONER_ID, referralRequest.getRequester().getAgent().getReference()),
+            () -> assertEquals("Organization/9C3AB881-FCDE-48EC-8028-37B20E0AE893",
+                               referralRequest.getRecipient().get(0).getReference()),
+            () -> assertEquals("Reason Code 1", referralRequest.getReasonCodeFirstRep().getCoding().get(0).getDisplay())
+        );
+    }
+
     @Test
     public void mapReferralRequestWithValidData() {
 
         var codeableConcept = createCodeableConcept(REASON_CODE_1, SNOMED_SYSTEM, CODING_DISPLAY);
         when(codeableConceptMapper.mapToCodeableConcept(any()))
                 .thenReturn(codeableConcept);
 
-        var ehrComposition = unmarshallEhrCompositionElement("full_valid_data_example.xml");
+        var ehrExtract = unmarshallEhrExtractElement("full_valid_data_example.xml");
```

Comment:
Wait, I'm so confused as to why this example is referencing a Practitioner on line 109/110. The XML shows no person involved?!!

Reply:
I will look into this, I didn't change this test in essence, just added some extra xml wrappers to align it with other tests...

Reply:
I corrected full_valid_data_example.xml and it includes 58341512-03F3-4C8E-B41C-A8FCA3886BBB (PRACTITIONER_ID) which was there originally, so now it is checking for valid PRACTITIONER_ID